src/Core/StoreOptions.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Identity
+{
+    /// <summary>
+    /// Used for store specific options
+    /// </summary>
+    public class StoreOptions
+    {
+        /// <summary>
+        /// If set to a positive number, the default OnModelCreating will use this value as the max length for any 
+        /// properties used as keys, i.e. UserId, LoginProvider, ProviderKey.
+        /// </summary>
+        public int MaxLengthForKeys { get; set; }
```

Comment:
This is very EF specific, should this be in Core?

Reply:
This one is kinda EF specific, but this is in Core because we are going to be adding the ProtectUserData stuff here as well, and Core will need to enforce something when the new flag is true...  

I was going to have StoreOptions be stand alone in the .Stores package if not for the next PR